gh-71316: Update dis documentation to include changes to jump arguments (#95798)
python/cpython: Doc/library/dis.rst

      Use 2 bytes for each instruction. Previously the number of bytes varied
      by instruction.

   .. versionchanged:: 3.10

>       The argument for instructions that have a jump now use the instruction
>       offset instead of the byte offset. Because each instruction take two
>       bytes, the instruction offset is exactly half of the byte offset.

Reviewer: ```suggestion
      The argument of jump instructions is now the instruction offset rather than
      the byte offset.
```

Author: This does not only affect jump instructions, but instructions that has a potential jump (in particular, in additional to all the jump instructions, `SETUP_WITH`, `FOR_ITER` and `SETUP_FINALLY`). In general, the instructions affected are the one whose argument name in the documentation is `target` or `delta`. Maybe this?
```suggestion
      The argument of jump, exception handling and loop instructions is now
      the instruction offset rather than the byte offset.
```

Reviewer: LGTM.
